packages/ckeditor5-engine/src/view/renderer.ts

```diff
@@ -676,21 +695,45 @@ export default class Renderer extends Observable {
 		// and it disrupts the whole algorithm. See https://github.com/ckeditor/ckeditor5/issues/6367.
 		//
 		// It doesn't matter in what order we remove or add nodes, as long as we remove and add correct nodes at correct indexes.
-		for ( const action of diff ) {
+		for ( const action of actions ) {
 			if ( action === 'delete' ) {
+				// @if CK_DEBUG_TYPING // if ( window.logCKETyping ) {
+				// @if CK_DEBUG_TYPING // 	console.info( '%c[Renderer]%c Remove node',
+				// @if CK_DEBUG_TYPING // 		'color: green;font-weight: bold', '', actualDomChildren[ i ]
+				// @if CK_DEBUG_TYPING // 	);
+				// @if CK_DEBUG_TYPING // }
 				nodesToUnbind.add( actualDomChildren[ i ] as DomElement );
 				remove( actualDomChildren[ i ] );
-			} else if ( action === 'equal' ) {
+			} else if ( action === 'equal' || action === 'replace' ) {
 				i++;
 			}
 		}
 
 		i = 0;
 
-		for ( const action of diff ) {
+		for ( const action of actions ) {
 			if ( action === 'insert' ) {
+				// @if CK_DEBUG_TYPING // if ( window.logCKETyping ) {
+				// @if CK_DEBUG_TYPING // 	console.info( '%c[Renderer]%c Insert node',
+				// @if CK_DEBUG_TYPING // 		'color: green;font-weight: bold', '', expectedDomChildren[ i ]
+				// @if CK_DEBUG_TYPING // 	);
+				// @if CK_DEBUG_TYPING // }
+
 				insertAt( domElement as DomElement, i, expectedDomChildren[ i ] );
 				i++;
+			} else if ( action === 'replace' ) {
+				// @if CK_DEBUG_TYPING // if ( window.logCKETyping ) {
+				// @if CK_DEBUG_TYPING // 	console.group( '%c[Renderer]%c Update text node',
+				// @if CK_DEBUG_TYPING // 		'color: green;font-weight: bold', ''
+				// @if CK_DEBUG_TYPING // 	);
+				// @if CK_DEBUG_TYPING // }
+
+				updateTextNode( actualDomChildren[ i ] as DomText, ( expectedDomChildren[ i ] as DomText ).data );
```

Comment:
AFAICS this method so far was only touching elements but now it also updates text nodes. This only happens on Android, right? If so:

* Add info about what this method does for Android to its documentation.
* Document this `else if()` that it's Android only.